src/dfx/src/commands/quickstart.rs

```diff
@@ -0,0 +1,129 @@
+use anyhow::{bail, Context};
+use candid::Principal;
+use dialoguer::{Confirm, Input};
+use ic_utils::interfaces::{
+    management_canister::builders::InstallMode, ManagementCanister, WalletCanister,
+};
+use indicatif::ProgressBar;
+use num_traits::Inv;
+use rust_decimal::Decimal;
+use tokio::runtime::Runtime;
+
+use crate::{
+    commands::ledger::{create_canister::MEMO_CREATE_CANISTER, notify_create, transfer_cmc},
+    lib::{
+        environment::Environment,
+        error::DfxResult,
+        identity::Identity,
+        ledger_types::{Memo, NotifyError},
+        nns_types::{
+            account_identifier::AccountIdentifier,
+            icpts::{ICPTs, TRANSACTION_FEE},
+        },
+        operations::{
+            canister::install_wallet,
+            ledger::{balance, xdr_permyriad_per_icp},
+        },
+        provider::create_agent_environment,
+        waiter::waiter_with_timeout,
+    },
+    util::{assets::wallet_wasm, expiry_duration},
+};
+
+pub fn exec(env: &dyn Environment) -> DfxResult {
+    let env = create_agent_environment(env, Some("ic".to_string()))?;
+    let agent = env.get_agent().expect("Unable to create agent");
+    let ident = env.get_selected_identity().unwrap();
+    let principal = env.get_selected_identity_principal().unwrap();
+    eprintln!("Your DFX user principal: {principal}");
+    let acct = AccountIdentifier::new(principal, None);
+    eprintln!("Your ledger account address: {acct}");
+    let runtime = Runtime::new().expect("Unable to create a runtime");
+    runtime.block_on(async {
+        let balance = balance(agent, &acct, None).await?;
```

Comment:
Can we break this up into some smaller methods?  The intent would be to increase readability since there are so many nested if-else-if levels.

Reply:
I do not think we can. Each step has several outputs that are all used in later steps. If readability is an issue I can certainly add comments though.